py4web/utils/grid.py

```diff
@@ -1124,6 +1131,11 @@ def make_action_buttons(self, row):
                         # if None was returned, no button is available for this row: ignore this value in the
                         # list
                         continue
+                attrs = (
+                    self.attributes_plugin.confirm(message=self.T(btn.message))
```

Comment:
My understanding was that self.T in this case is not actually a Translation object and that it didn't add any requirements to Grid.

The developer can pass a value to self.T on the grid call

`T=lambda text: text,`

and the default behavior is just to return the actual text.

I only included it because that is what is done on the Delete confirmation button.  However, I can also see that it would be nice to allow that btn.message to be a callable so that the developer could craft a more specific message using some values from the row. Then I think having translation inside Grid in this manner would be helpful.

The use case for allowing a confirmation on these action buttons is that I want to have a button on the row to perform a quick action, say activate or deactivate an entity, This works just fine, but it would be nice to pop a confirmation message before the action is performed.

Of  course it's possible that I'm missing something here too. 

I will go ahead and remove if you think this is a bad practice.

Reply:
I will merge as is and then audit the grid to make sure this is handled consistently.